websockets/protocol/caret.go

```diff
@@ -0,0 +1,81 @@
+package protocol
+
+type Caret struct {
+	Start int `json:"caret_start"`
+	End   int `json:"caret_end"`
```

Comment:
Also can you change the JSON fields to just be `start` and `end`?

Reply:
Damn, so many flaws.

Reply:
Lol yeah I spoke too soon.